mnemonic-memory-services/mnemonic-nvml-vmem-service/src/main/native/org_apache_mnemonic_service_allocatorservice_internal_VMemServiceImpl.c

```diff
@@ -134,6 +167,48 @@ void JNICALL Java_org_apache_mnemonic_service_allocatorservice_internal_VMemServ
 }
 
 JNIEXPORT
+void JNICALL Java_org_apache_mnemonic_service_allocatorservice_internal_VMemServiceImpl_nsetHandler(
+    JNIEnv *env, jobject this, jlong id, jlong key, jlong value)
+{
+  pthread_rwlock_rdlock(&g_vmem_rwlock);
+  pthread_mutex_lock(g_vmem_mutex_ptr + id);
+  void *md = *(g_vmp_ptr + id);
+  if (id < PMALLOC_KEYS && id >= 0) {
+    pmalloc_setkey(md, key, (void*)value);
```

Comment:
Please disable the setkey()/getkey() temporarily and throw exceptions if used by developers , we could add this support in another task.